samples/ustream_split/src/main.c

```diff
@@ -0,0 +1,111 @@
+
+    //Change return to AZ_ULIB_SUCCESS if last returned value was AZ_ULIB_EOF
+    if(result == AZ_ULIB_EOF)
+    {
+        result = AZ_ULIB_SUCCESS;
+    }
+    return result;
+}
+
+/**
+ * This sample creates a ustream and then splits it into two ustreams at a desired position.
+ * The following steps are followed:
+ *      1. Create the ustream and print the ustream to the console. It needs to be reset
+ *          before splitting so that the current position is restored to the beginning.
+ *      2. Split the ustream at the desired position.
+ *      3. Print each ustream to the console.
+ *      4. Dispose of the two ustreams.
+ */
+int main(void)
+{
+    AZ_ULIB_RESULT result;
+
+    AZ_USTREAM_DATA_CB *data_cb;
+    if((data_cb = (AZ_USTREAM_DATA_CB *)malloc(sizeof(AZ_USTREAM_DATA_CB))) != NULL)
+    {
+        AZ_USTREAM ustream_instance;
+        if((result = az_ustream_init(&ustream_instance, data_cb, free, (const uint8_t*)USTREAM_ONE_STRING, 
+                                                        sizeof(USTREAM_ONE_STRING), NULL)) != AZ_ULIB_SUCCESS)
+        {
+            printf("Could not initialize ustream_instance\r\n");
+        }
+        else if((result = print_ustream(&ustream_instance)) != AZ_ULIB_SUCCESS)
+        {
+            az_ustream_dispose(&ustream_instance);
+            printf("Could not print the original ustream_instance\r\n");
+        }
+        else if((result = az_ustream_reset(&ustream_instance)) != AZ_ULIB_SUCCESS)
+        {
+            az_ustream_dispose(&ustream_instance);
+            printf("Could not reset ustream_instance\r\n");
+        }
+        else
+        {
+            AZ_USTREAM ustream_instance_split;
+
+            if((result = az_ustream_split(&ustream_instance, &ustream_instance_split, SPLIT_POSITION)) != AZ_ULIB_SUCCESS)
+            {
+                az_ustream_dispose(&ustream_instance);
+                printf("Could not split ustream_instance\r\n");
+            }
+            else if((result = print_ustream(&ustream_instance)) != AZ_ULIB_SUCCESS)
+            {
+                az_ustream_dispose(&ustream_instance);
+                az_ustream_dispose(&ustream_instance_split);
+                printf("Could not print the split ustream_instance\r\n");
+            }
+            else if((result = print_ustream(&ustream_instance_split)) != AZ_ULIB_SUCCESS)
+            {
+                az_ustream_dispose(&ustream_instance);
+                az_ustream_dispose(&ustream_instance_split);
+                printf("Could not print ustream_instance_split\r\n");
+            }
+            else if((result = az_ustream_dispose(&ustream_instance)) != AZ_ULIB_SUCCESS)
+            {
+                az_ustream_dispose(&ustream_instance_split);
+                printf("Could not dispose of ustream_instance\r\n");
+            }
+            else if((result = az_ustream_dispose(&ustream_instance_split)) != AZ_ULIB_SUCCESS)
```

Comment:
Use only one dispose  for ustream_instance_split at the end of this else.

Reply:
done. cleaned this up a bit

---
In reply to: [333763442](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r333763442) [](ancestors = 333763442)